src/api/extendobservable.ts

```diff
@@ -67,7 +67,8 @@ export function extendObservableObjectWithProperties(
         )
         if (decorators) {
             const keys = getPlainObjectKeys(decorators)
-            for (const key of keys) {
+            for (let i = 0; i < keys.length; i++) {
```

Comment:
Why not `getPlainObjectKeys(decorators).forEach`? Same for the other loop. This is probably variant nobody has considered yet? I don't see benefit much of benefit for such manual loop.

Reply:
I don't think performance or readability are in contention here; it's an idiomatic for loop which squeezes the most perf (even a small amount is important for a lib like mobx, right?) and is common enough not to present a readability challenge afaik.